sbibm/tasks/norefposterior/task.py

```diff
@@ -0,0 +1,365 @@
+            self.min_axis, self.max_axis
+        ).detach()
+
+    def get_prior(self) -> Callable:
+        def prior(num_samples=1):
+            return pyro.sample("parameters", self.prior_dist.expand_by([num_samples]))
+
+        return prior
+
+    def get_simulator(self, max_calls: Optional[int] = None) -> Simulator:
+        """Get function returning samples from simulator given parameters
+
+        Args:
+            max_calls: Maximum number of function calls. Additional calls will
+                result in SimulationBudgetExceeded exceptions. Defaults to None
+                for infinite budget
+
+        Return:
+            Simulator callable
+        """
+
+        def simulator(parameters):
+            num_samples = parameters.shape[0]
+
+            m_ = torch.stack(
+                (parameters[:, [0]].squeeze(), parameters[:, [1]].squeeze())
+            ).T
+            if m_.dim() == 1:
+                m_.unsqueeze_(0)
+
+            m = torch.broadcast_to(m_, (self.max_axis, self.max_axis, *m_.shape))
+
+            s1 = parameters[:, [2]].squeeze()  # ** 2
+            s2 = parameters[:, [3]].squeeze()  # ** 2
+
+            # Note: checking the covariance_matrix for valid inputs
+            #       (being positive semidefinite) is expense, so
+            #       `S` needs to be PSD compliant
+            #       for the future: consider rotating img for more variability
+            S = torch.empty((self.max_axis, self.max_axis, num_samples, 2, 2))
+            S[..., 0, 0] = s1 ** 2
+            S[..., 0, 1] = s1 * s2
+            S[..., 1, 0] = 0.0  # s1 * s2
+            S[..., 1, 1] = s2 ** 2
+
+            # Add eps to diagonal to ensure PSD
+            eps = 0.000001
+            S[..., 0, 0] += eps
+            S[..., 1, 1] += eps
+
+            assert S.shape == (
+                self.max_axis,
+                self.max_axis,
+                num_samples,
+                2,
+                2,
+            ), f"{name_display} :: cov matrix {S.shape} != expectation"
+            assert m.shape == (
+                self.max_axis,
+                self.max_axis,
+                num_samples,
+                2,
+            ), f"{name_display} :: mean vector {m.shape} != expectation"
+
+            # define the probility distribution of our beamspot
+            # on a 2D grid (in batches)
+            data_dist = pdist.MultivariateNormal(m.float(), S.float(),
+                                                 # `S` is constructed positive semidefinite
+                                                 # validation is expensive
+                                                 validate_args = False
+                                                 )
+
+            valb = bcast_coordinate_field(
+                self.base_coordinate_field, num_samples
+            ).detach()
+            # valb = quadratic_coordinate_field(self.min_axis, self.max_axis, num_samples).detach()
```

Comment:
Can be removed?

Reply:
:face_with_head_bandage: Well spotted. Fixed with 830f4c5